### PR TITLE
Full Changesets usage to automate releases

### PR DESCRIPTION
**Pull request by leordev**
Completes the release pipeline which fully uses Changesets to publish to NPM, create git tags and releases with proper changelog notes.

Working evidences in a forked repo:
- Git tags for each package publication with notes:  https://github.com/leordev/tbdex-js-releases/tags
  - GH Releases are also mirrored for free: https://github.com/leordev/tbdex-js-releases/releases
- Release pipeline generated from a regular main push, generating a Version Packages PR, skipping npm packages publication: 
  - https://github.com/leordev/tbdex-js-releases/actions/runs/7895015042
  - https://github.com/leordev/tbdex-js-releases/pull/6
- Release pipeline generated from the merge of a Version Packages PR, which publish to NPM and publish docs: 
  - https://github.com/leordev/tbdex-js-releases/actions/runs/7895056061 (please ignore the GH Pages deploy because I didnt enable that for my personal forked repo)

I've also updated the README release section.

Also, if there are any present changesets on `main` we are automatically publishing a preview release using the `next` tag.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -89,27 +89,27 @@ Prior to merging your branch into main, and given you have relevant semantic ver
 
 **You can stop here!** It is recommended to merge your branch into main with the `.changeset/*.md` files, at which point, the Changeset GitHub Action will automatically pick up those changes and open a PR to automate the `pnpm changeset version` execution. For example, [see this PR](https://github.com/TBD54566975/tbdex-js/pull/36). This command will do two things: update the version numbers in the relevant `package.json` files & also aggregate Summary notes into the relevant `CHANGELOG.md` files. In keeping with the staged commit analogy, this is akin to the actual commit.
 
-## Cutting Releases
+## Publishing Releases
 
-When a changeset PR is merged to main we will automatically create a GitHub release using the workflow [Create GH Release](./.github/workflows/create-gh-release.yml).
+Just merge the [Version Packages PR](https://github.com/TBD54566975/tbdex-js/pulls?q=is%3Apr+author%3Aapp%2Fgithub-actions+%22Version+Packages%22+) when you are ready to publish the new versions!
```
Comment (KendallWeihe) [inline README.md:94]: nice, this language is much better!

File: package.json [hunk 2]
```diff
@@ -21,20 +21,23 @@
...
+    "typescript": "5.2.2"
   },
   "pnpm": {
     "overrides": {
       "browserify-sign@>=2.6.0 <=4.2.1": ">=4.2.2",
       "follow-redirects": ">=1.15.4"
     }
+  },
+  "dependencies": {
+    "@changesets/changelog-github": "^0.5.0"
```
Comment (KendallWeihe) [inline package.json:41]: could this be a dev dependency?

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        id: changesets
+        with:
+          publish: pnpm changeset publish
+          version: pnpm changeset version
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+      - name: Publish `next` preview
```
Comment (KendallWeihe) [inline .github/workflows/release.yml:67]: so what does this do exactly? it published a package with version equal to `next`?

what is this checking for (line below) `steps.changesets.outputs.pullRequestNumber != ''`

File: package.json (hunk 2 above)
Comment (KendallWeihe) [inline package.json:41]: could this be a dev dependency?

File: .github/workflows/release.yml (hunk 3 above)
Comment (KendallWeihe) [reply .github/workflows/release.yml:67]: oops, didn't read the full description, my bad 

"Also, if there are any present changesets on main we are automatically publishing a preview release using the next tag."

excellent!

**Review by KendallWeihe — APPROVED** [review]
very cool! 

one comment about dev dependencies, but lgtm! ✅

File: package.json (hunk 2 above)
Comment (leordev) [reply package.json]: Already fixed!

Comment (leordev) [reply package.json]: Already fixed!

File: .github/workflows/release.yml (hunk 3 above)
Comment (leordev) [reply .github/workflows/release.yml:67]: Thanks! Leaving this unresolved for awareness!

**Review by jiyoontbd — APPROVED** [review]

**Pull request merged**